include/bluetooth/l2cap.h

```diff
@@ -150,6 +150,22 @@ struct bt_l2cap_chan {
 #endif /* CONFIG_BT_L2CAP_DYNAMIC_CHANNEL */
 };
 
+/** @brief Get @c chan->state.
+ *
+ * This field does not exist when @kconfig{CONFIG_BT_L2CAP_DYNAMIC_CHANNEL} is
+ * disabled. In that case, this function returns @ref BT_L2CAP_CONNECTED since
+ * the struct in that case can only represent static channels and static
+ * channels are always connected.
+ */
+static inline bt_l2cap_chan_state_t bt_l2cap_chan_get_state(struct bt_l2cap_chan *chan)
```

Comment:
is it expected to be used by applications?   (I'd assume that app should rely disconnected callback instead of tracking channel state)

Reply:
That's a fair question. If there's no obvious good use case for external usage it should probably be moved to `l2cap_internal.h`.

Reply:
Alternative, this could be moved to a `l2cap_info` struct and be moved to a more generic function such as `bt_l2cap_get_info` similar to `bt_conn_get_info` and `bt_iso_get_info`, so that we can easily extend a single function with this information. 

That being said, since the `struct bt_l2cap_chan` is a public struct, this value is already available to the application without this function.

Reply:
I think this is a really innocent function. I don't see how keeping it can shoot us in the foot later. 

> If there's no obvious good use case for external usage it should probably be moved to l2cap_internal.h.

IMO getters, setters, constructors and such should be next to their struct. It's good for discoverability. If they are hidden away in other files, it's easy to not notice them.

> We could move the state so that it is no longer guarded by CONFIG_BT_L2CAP_DYNAMIC_CHANNEL if we want to expose it to the application for static L2CAP channels.

It's possible. I don't have any strong feelings for or against this. (But I would prefer not to have to change the PR.) Anyone have strong feelings?

Reply:
> I think this is a really innocent function. I don't see how keeping it can shoot us in the foot later.

I don't see any reason to add a function that doesn't add any new functionality or information to a public API. 

> Anyone have strong feelings?

Not sure I would call it strong, but if we add such as function, we may as well be consistent with the `bt_conn` and `bt_iso` APIs and implement it as a get_info function, IMO.

Reply:
I'll move the function to `l2cap.c` where it's used. That sounds good to me.